boxsdk/client/client.py

```diff
@@ -539,6 +539,251 @@ def create_group(self, name):
+        :rtype:
+            :class:`FileVersionRetention`
+        """
+        return self.translator.translate('file_version_retention')(session=self._session, object_id=retention_id)
+
+    def retention_policy_assignment(self, assignment_id):
+        """
+        Initialize a :class:`RetentionPolicyAssignment` object, whose box id is assignment_id.
+
+        :param assignment_id:
+            The box ID of the :class:`RetentionPolicyAssignment` object.
+        :type assignment_id:
+            `unicode`
+        :return:
+            A :class:`RetentionPolicyAssignment` object with the given assignment ID.
+        :rtype:
+            :class:`RetentionPolicyAssignment`
+        """
+        return self.translator.translate('retention_policy_assignment')(session=self._session, object_id=assignment_id)
+
+    def create_retention_policy(
+            self,
+            policy_name,
+            disposition_action,
+            retention_length,
+            can_owner_extend_retention=None,
+            are_owners_notified=None,
+            custom_notification_recipients=None,
+    ):
+        """
+        Create a retention policy for the given enterprise.
+
+        :param policy_name:
+            The name of the retention policy.
+        :type policy_name:
+            `unicode`
+        :param retention_length:
+            The amount of time in days to apply the retention policy to the selected content.
+            Do not specify for `indefinite` policies, only for `finite` policies.
+        :type retention_length:
+            `int` or float('inf')
+        :param disposition_action:
+            For `finite` policy can be set to `permanently delete` or `remove retention`.
+            For `indefinite` policy this must be set to `remove_retention`
+        :type disposition_action:
+            `unicode`
+        :param can_owner_extend_retention:
+            The owner of a file will be allowed to extend the retention if set to true.
+        :type can_owner_extend_retention:
+            `boolean` or None
+        :param are_owners_notified:
+            The owner or co-owner will get notified when a file is nearing expiration.
+        :type are_owners_notified:
+            `boolean` or None
+        :param custom_notification_recipients:
+            A custom list of user mini objects that should be notified when a file is nearing expiration.
+        :type custom_notification_recipients:
+            `list` or `User` objects
+        :return:
+            The newly created Retention Policy
+        :rtype:
+            :class:`RetentionPolicy`
+        """
+        url = self.get_url('retention_policies')
+        retention_attributes = {'policy_name': policy_name}
+        if retention_length == float('inf') or retention_length == -1:
+            retention_attributes['policy_type'] = 'indefinite'
+            retention_attributes['disposition_action'] = 'remove_retention'
```

Comment:
We should still pass the `disposition_action` the user passed in here instead of silently overriding it.